### ISSUE TITLE
Issue printing documents

### ISSUE THREAD

**Issue by pedrojrivera** [stephanrauh/ngx-extended-pdf-viewer#148]
**Check this first**
- Does the bug also happen on https://mozilla.github.io/pdf.js/web/viewer.htm?
NO
- Does the bug also happen on https://pdfviewer.net/simple?
NO

**Describe the bug**
When printing documents, the printing gets aborted. In chrome the print process gets aborted after the pages are rendered and print dialog never displays. In FF the first page is sent to print and I get a print dialog with only the first page.

I put the program in debug and viewer.js line 12696 abort() is always invoked in Chrome. 

```typescript
    let activeServiceOnEntry = activeService;
    activeService.renderPages().then(function () {
      return activeServiceOnEntry.performPrint();
    }).catch(function () {}).then(function () {
      if (activeServiceOnEntry.active) {
        abort();   **<= This always gets invoked!!**
      }
    });
```

**Version info**
- Version of ngx-extended-pdf-viewer
1.5.2
- Version of the JavaScript files pdf.js, pdf-worker.js and viewer.js (currently: default files vs. version 2.2 preview files)
2.2.228

**Desktop (please complete the following information):**
Windows 10

Chrome Version 77.0.3865.90 (Official Build) (64-bit)
Firefox 68.0.2 (64-bit)

**Additional context**
Angular version:  8.2
Angular Material 8.2

I suspect that it might have something to do with Material???? We have other apps and the issue doesn't exist. Could it be an issue with overlays???

**Comment by pedrojrivera** [comment]
#[
<img width="1172" alt="debug" src="https://user-images.githubusercontent.com/32545483/66256517-8112a100-e75c-11e9-80f1-897b4175de67.png">
](url)

**Comment by stephanrauh** [comment]
Strange. There's an exception, but as far as I can see, it's silently swallowed. Do you see it on the console window?

Currently, I don't know how to reproduce the bug, so I need your help. My showcase also uses Angular Material. It might me an incompatibility with Angular Material, but it's not simply a case of adding Material and seeing the bug. Can you give me a clue what's different between the buggy application and your other applications?

**Comment by pedrojrivera** [comment]
I am not getting any console messages. Here is my console at the break point of abort():

<img width="561" alt="console" src="https://user-images.githubusercontent.com/32545483/66268051-12d6e880-e807-11e9-9b27-6ef442823c7e.png">

Here is the activeServiceEntry object values. It looks at if active = true to determine if it should abort. What would cause this condition? 

<img width="360" alt="console2" src="https://user-images.githubusercontent.com/32545483/66268073-7eb95100-e807-11e9-9b5e-df2ccfe6ba02.png">

Here is a screen shot of the rendered html while on the breakpoint:

<img width="493" alt="console4" src="https://user-images.githubusercontent.com/32545483/66268111-099a4b80-e808-11e9-9889-e968fcefb57b.png">

I checked the printContainer div and it has all the rendered pages. 

The pdf viewer component is used inside of a multi-tab application and we do have our own tool bar implementation. So we tried reverting back to the default native pdfjs toolbar and it made no difference. 

<img width="777" alt="console5" src="https://user-images.githubusercontent.com/32545483/66268273-ca6cfa00-e809-11e9-85c2-f0ad7c24fe93.png">

I am going to compare the rendered html output from your examples to mine and see if there is anything visually different in the DOM structure. 

Let me know if there is anything I can provide.

**Comment by pedrojrivera** [comment]
@stephanrauh I discovered the issue and was able to fix the printing.

So I compared my application to yours and ran them side by side comparing every single line in the DOM and css attributes. 

I discovered that a certain css combination causes the issue. I made some css changes to your examples and was able to replicate the issue.

Here is the css code that triggers the problem:

```

<style>
body {
    overflow: hidden;
    height: 100vh;
}
html {
    overflow-y: hidden;
}
</style>

```

Once I removed the **overflow-y: hidden;** from the html section, the printing started to work as expected.

**Comment by pedrojrivera** [comment]
I was also able to replicate the error by simply adding those css entries at the pdfjs demo site.

https://mozilla.github.io/pdf.js/web/viewer.html

``` typescript
body, input, button, select {
    font: message-box;
    outline: none;
    overflow: hidden;  // Added
    height: 100vh;   // Added
}
html {
    height: 100%;
    width: 100%;
    font-size: 10px;
    overflow: hidden;  //Added -- removing this line fixes the issue
} ```

By doing this only the first page gets sent to the printer and all the other pages get ignored.

**Comment by stephanrauh** [comment]
I've managed to reproduce the bug. It shows slightly different on my machine, but the general gist is the same here: `overflow-y: hidden` causes trouble.

As a workaround, I'd use a media query like this:

```css
@media not print {
  html {
    overflow-y: hidden;
  }
}
```

**Comment by stephanrauh** [comment]
BTW, the `abort()` method doesn't seem to be called in my project. The problem seems to be simpler: the print algorithm simply hides everything but the PDF document and calls the standard `window.print()` method of JavaScript. `overflow-y: hidden` means there's merely half a page on the screen, so that's what's printed.

**Comment by pedrojrivera** [comment]
@stephanrauh Thanks.

**Comment by stephanrauh** [comment]
It seems to be safe to add the CSS rule to the component. That'll be part of version 1.5.3 (unless a reason why this might cause problems crops up).

**Comment by stephanrauh** [comment]
Published with version 1.5.3.

**Comment by stephanrauh** [comment]
I've went over the print CSS again. When you update to version 9.0.0, brace yourself for breaking changes. In your case, I don't expect them, but you never know.
